FortnitePorting/Plugins/Blender/import_task.py

```diff
@@ -574,6 +600,128 @@ def get_meta(search_props):
+                    if scale != Vector((1, 1, 1)):
+                        if entry.get('BoneName', '').casefold() == 'faceattach':
+                            face_attach_scale = scale
+                        Log.warn(f"Non-zero scale: {entry}")
+
+                if not shape_keys:
+                    # Create Basis shape key
+                    imported_mesh.shape_key_add(name="Basis", from_mix=False)
+
+                # NOTE: I think faceAttach affects the expected location
+                # I'm making this assumption from observation of an old
+                # export of face poses for Polar Patroller.
+                loc_scale = (0.01 if self.options.get("ScaleDown") else 1)
+                for pose in pose_data:
+                    # If there are no influences, don't bother
+                    if not (influences := pose.get('Keys')):
+                        continue
+
+                    if not (pose_name := pose.get('Name')):
+                        Log.warn("skipping pose data with no pose name")
+                        continue
+
+                    # Enter pose mode
+                    bpy.context.view_layer.objects.active = armature
+                    bpy.ops.object.mode_set(mode='POSE')
+
+                    # Reset all transforms to default
+                    bpy.ops.pose.select_all(action='SELECT')
+                    bpy.ops.pose.transforms_clear()
+                    bpy.ops.pose.select_all(action='DESELECT')
+
+                    # Move bones accordingly
+                    contributed = False
+                    for bone in influences:
+                        if not (bone_name := bone.get('Name')):
+                            Log.warn(f"empty bone name for pose {pose}")
+                            continue
+
+                        pose_bone: bpy.types.PoseBone = get_case_insensitive(armature.pose.bones, bone_name)
+                        if not pose_bone:
+                            # For cases where pose data tries to move a non-existent bone
+                            # i.e. Poseidon has no 'Tongue' but it's in the pose asset
+                            if is_head:
+                                # There are likely many missing bones in non-Head parts, but we
+                                # process as many as we can.
+                                Log.warn(f"could not find: {bone_name} for pose {pose_name}")
+                            continue
+
+                        # Check if there's a vertex group associated with the pose bone
+                        # If not, no reason to pose it.
+                        if pose_bone.name not in imported_mesh.vertex_groups:
+                            continue
+
+                        # Reset bone to identity
+                        pose_bone.matrix_basis.identity()
+
+                        rotation = bone.get('Rotation')
+                        assert rotation.get('IsNormalized'), "non-normalized rotation unsupported"
```

Comment:
cant we just normalize the rotations during the export process if they're not normalized

Reply:
Done.